lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb

```diff
@@ -44,6 +48,12 @@ def run_with_tracing(*args)
 
           private
 
+          def process_distributed_tracing_headers(active_call)
+            return unless active_call.metadata
```

Comment:
Out of curiosity, have you found any cases where active_call.metadata was absent?

Reply:
I haven't. Given that the underlying gRPC source code (an obvious port) isn't particularly strict about type checking or input validation, I wanted to be as cautious as possible. But now that you mention it, I see that we do not check `active_call` itself for `nil`, which is far more likely I'd imagine, and we also don't perform this same check when we're spawning a `finishable` segment. I'll make some updates.

Reply:
With [cb8b5dc](https://github.com/newrelic/newrelic-ruby-agent/pull/1286/commits/cb8b5dcb08e4b3aadddd810b8a6b96782854033c) I've added a new method for obtaining the metadata that lets us be consistent about it.